lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -419,11 +419,9 @@ impl<TGraphLinks: GraphLinks> HNSWIndex<TGraphLinks> {
         {
             for (idx, deleted) in deleted_bitslice.iter().enumerate() {
                 if *deleted {
-                    debug_assert!(graph
-                        .links
-                        .links(idx as PointOffsetType, 0)
-                        .next()
-                        .is_none());
+                    graph.links.for_each_link(idx as PointOffsetType, 0, |_| {
+                        panic!("Deleted point in the graph");
```

Comment:
I'd prefer to keep a debug-level panic here. You never know.

Reply:
It's behind `#[cfg(debug_assertions)]`, check 9 lines above.

Reply:
Thanks! The GitHub diff hid it from me :face_holding_back_tears: